src/ui/reducers/possibleBreakpoints.ts

```diff
@@ -120,17 +125,21 @@ export const getPossibleBreakpointsForSelectedSource = (state: UIState): Locatio
   return getPossibleBreakpointsForSource(state, sourceId) || [];
 };
 
-export const getBreakableLinesForSource = (state: UIState, sourceId: string) => {
-  return uniq(
-    adapterSelectors
-      .selectById(state, sourceId)
-      ?.possibleBreakpoints?.map(location => location.line)
-  );
-};
+export const getBreakableLinesForSource = createSelector(
+  (state: UIState) => state.possibleBreakpoints,
+  (_state: UIState, sourceId: string): string => sourceId,
+  (possibleBreakpointsSlice, sourceId: string) => {
+    const possibleBreakpoints = possibleBreakpointsSlice.entities[sourceId];
```

Comment:
:hand: realistically we could actually move this line into line 129 - we have both `state` and `sourceId` available.

We _could_ also give this selector a larger cache size, although I'm not sure if it's absolutely necessary given that we really just use this for the selected source.